Initial setup

ranjan-stha opened the pull request:
Added the modules for text extraction and model prediction
Added the Terraform configuration for setting up the SQS, lambda functions, API Gateway  for both text extraction and model predictions.
Added the mock server for local testing the integration.

ranjan-stha committed:
main terraform setup for modules;

ranjan-stha committed:
modules added for apigateway, sqs and lambda

ranjan-stha committed:
lambda fns added for handling input reqs, predict_entry, extract_docs;

ranjan-stha committed:
lambda fn config added for sending request towards deep;

ranjan-stha committed:
callback_url added and queued;

ranjan-stha committed:
added callback_url in sqs in initial request handling;

ranjan-stha committed:
added lambda for generating signed urls and sending request(wip)

ranjan-stha committed:
handles and puts back msg in the sqs when response ok not received;

ranjan-stha committed:
added deadletter queue to handle msg which are not processed after fixed number of attempts;

ranjan-stha committed:
directory renamed;

ranjan-stha committed:
path fixes;

ranjan-stha committed:
msg updated in response;

ranjan-stha committed:
added mock server; updated lambda fn to handle during local tests;

ranjan-stha committed:
check for the content type of the url;

ranjan-stha committed:
s3 bucket added for storing text extracted docs;

ranjan-stha committed:
added s3 bucket policy;

ranjan-stha committed:
client configs added;

ranjan-stha committed:
use a default aws region if not found;

ranjan-stha committed:
docker related files added for local testing;

ranjan-stha committed:
host updated;

ranjan-stha committed:
env variable added;

ranjan-stha committed:
mock response updated; key-value updated

bimal125 committed:
Change server port

ranjan-stha committed:
sends individual request on callback url;

ranjan-stha committed:
added a fake total_pages field in the request;

bimal125 committed:
Fix url content type

bimal125 committed:
Add flake8 file

ranjan-stha committed:
added a route to read file/images from local dir;

bimal125 committed:
Change hostname

ranjan-stha committed:
domain name added to web extracted links;

ranjan-stha committed:
terraform plan github action config added;

ranjan-stha committed:
updated the terraform plan config;

ranjan-stha committed:
config fixes;

ranjan-stha committed:
single quote used;

ranjan-stha committed:
minor change;

ranjan-stha committed:
terraform apply config added;

ranjan-stha committed:
added the words_count and page_count (fake value added for now) in the response  until the parsing tool is updated;

ranjan-stha committed:
remove non-required imports;

ranjan-stha committed:
some flake8 fixes; added the fake data for model predictions; request/response added for model predictions

ranjan-stha committed:
added the url_content_type field in the request;

ranjan-stha committed:
added s3 backend and dynamodb tbl to track the tf states;

ranjan-stha committed:
added callback url for prediction endpoint; sends POST request on the callback url along with predictions and thresholds;

ranjan-stha committed:
not to delete the keys from the fake json data on local server setup;

ranjan-stha committed:
tags enums and mappings are added; endpoints are created/updated;

ranjan-stha committed:
moved the mapping function;

ranjan-stha committed:
handled requests for entry prediction on production and mock server;

ranjan-stha committed:
updated terraform configs on entry prediction; used lambda layers for accessing the mappings;

ranjan-stha committed:
added the requests lib in the requirements.txt

ranjan-stha committed:
constants added which contains the categories version;

ranjan-stha committed:
added the number of pages extracted via parsing tool

ranjan-stha committed:
updated the parsing tool commit id in the requirements;

ranjan-stha committed:
lambda layer name updated;

ranjan-stha committed:
run mockserver as module;

ranjan-stha committed:
updated the deepl-parser lib;

ranjan-stha committed:
queues and lambda fn configs added for entry prediction;

ranjan-stha committed:
input request handler for entry prediction lambda fn added;

ranjan-stha committed:
lambda fn to send predicted values to deep;

ranjan-stha committed:
lambda fn to put back to queue when message are not successfully delivered on callback fn;

ranjan-stha committed:
removed the entry prediction lambda fn from this module;

ranjan-stha committed:
lambda layer added to share the mappings;

ranjan-stha committed:
entry pred related module added;

ranjan-stha committed:
tags related mappings added;

ranjan-stha committed:
lambda function to send the request to sagemaker endpoint for prediction;

ranjan-stha committed:
handler name updated in entry pred module;

ranjan-stha committed:
celery task added in the mockserver for text extraction;

bimal125 committed:
Clean code

ranjan-stha committed:
implemented the func to get url content type

ranjan-stha committed:
added a bucket to store the deepl-parser as a lambda layer; updated sqs msg size;

ranjan-stha committed:
updated the paths to the mapping zip file which is a part of lambda layer;

ranjan-stha committed:
updated the extract_docs lambda function; also updated the celery task; removed deepl-parser from requirements as it is deployed as lambda layer;

ranjan-stha committed:
prediction status added;

ranjan-stha committed:
extraction status field added;

ranjan-stha committed:
moved the python zipped of mapping to another location;

ranjan-stha committed:
added the deepl-parser zipped version for lambda layer;

ranjan-stha committed:
timeouts updated for lambda fn and sqs queues;

ranjan-stha committed:
updated lambda fns with threading for sending async requests; includes requests on sagemaker endpoint and geolocation lambda fn;

ranjan-stha committed:
terraform version updated;

ranjan-stha committed:
variables setup for dev and prod environments with terraform workspaces;

ranjan-stha committed:
fixed the variables, env vars; set lambda fn  permissions on specific resources;

ranjan-stha committed:
updated the lambda func names in the configs;

ranjan-stha committed:
fake data added for predictions; handle the prediction response and sends it towards deep callback_url;

ranjan-stha committed:
use ecr image for deepl extractor tool instead of lambda layers;

ranjan-stha committed:
Docker file to build the image for deepl extractor tool;

ranjan-stha committed:
return list for geolocation response;

ranjan-stha committed:
added a script to initialize initial setups; updated Dockerfile to install private repo by injecting ssh key for local setup;

ranjan-stha committed:
renamed few directories;

ranjan-stha committed:
added pull request template;

tnagorra commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,24 @@
+Addresses xxxxxx (eg: #1 OR toggle-corp/hr-portal-server#1) \
+Depends on xxxxxx (eg: #1 OR toggle-corp/hr-portal-server#1)
```
```suggestion
Addresses xxxxxx
Depends on xxxxxx
```

tnagorra commented on Dockerfile:
```diff
@@ -0,0 +1,25 @@
+
+FROM python:3.8-slim-buster
+
+LABEL maintainer="ToggleCorp Dev dev@togglecorp.com"
```
```suggestion
LABEL maintainer="Deep Dev dev@thedeep.com"
```

tnagorra commented on Dockerfile:
```diff
@@ -0,0 +1,25 @@
+LABEL maintainer="ToggleCorp Dev dev@togglecorp.com"
+
+COPY . /code/
+WORKDIR /code
```
We should move this COPY to the end and only COPY what's required for pip install here

tnagorra commented on mappings/constants.py:
```diff
@@ -0,0 +1,10 @@
+SECTOR_VERSION = 0.1
+SUBPILLAR_VERSION = 0.1
+SECONDARY_TAGS_VERSION = 0.1
+SUBPILLARS_1D_VERSION = 0.1
+SUBPILLARS_2D_VERSION = 0.1
+AGE_VERSION = 0.1
+GENDER_VERSION = 0.1
+SPECIFIC_NEEDS_GROUP_VERSION = 0.1
+SEVERITY_VERSION = 0.1
+AFFECTED_GRP_VERSION = 0.1
```
Any reason we start our versions from 0.1?
Do we follow a specific versioning scheme?
@thenav56 @AdityaKhatri

thenav56 replied:
Also, let's use string instead of float for versions.

ranjan-stha replied:
done

tnagorra commented on mockserver/app.py:
```diff
@@ -0,0 +1,86 @@
+app = Flask(__name__)
+
+
+@app.route('/')
```
IMO all of the response should be in same format.
Some of the response are text while others are JSON

ranjan-stha replied:
done. all response in json.

tnagorra commented on mappings/tags_mapping.py:
```diff
@@ -0,0 +1,343 @@
+
+from .constants import (
+    SECTOR_VERSION,
+    SUBPILLAR_VERSION,
+    SECONDARY_TAGS_VERSION,
+    SUBPILLARS_1D_VERSION,
+    SUBPILLARS_2D_VERSION,
```
What do we mean by Subpillar?
We previously had sub-pillar but it was replaced with sub-sectors and sub-dimensions.

ranjan-stha replied:
@tnagorra : The output of the model predictions still use subpillar which contains tags from both subpillar1d and subpillar2d. These subpillars versions are used to track the if the updated tags are used to train new models later. Also, note that there are no sub-sector models trained until now.

ranjan-stha replied:
The pillars mappings are removed.

tnagorra commented on mappings/tags_mapping.py:
```diff
@@ -0,0 +1,343 @@
+    UNACCOMPANIED_OR_SEPARATED_CHILDREN = MainTags(id=412, key='specific_needs_groups->Unaccompanied or Separated Children', version=SPECIFIC_NEEDS_GROUP_VERSION)
+
+    # Gender
+    FEMALE = MainTags(id=501, key='gender_kw_pred->Female', version=GENDER_VERSION)
```
Do we need to define a certain scheme for key?
looks like a hybrid of snake_case and CamelCase

ranjan-stha replied:
@tnagorra : the model response uses such type of keys. Make sense to use the proper format. I will talk with Selim regarding this.

ranjan-stha replied:
I talked with Selim last week. There will be re-structuring of the tags.

tnagorra commented on mappings/tags_mapping.py:
```diff
@@ -0,0 +1,343 @@
+            Tags.PROTECTION.value._asdict(),
+            Tags.SHELTER.value._asdict(),
+            Tags.WASH.value._asdict()
+        ]
```
I think it would be easier to write `[t.value.__asdict() for t in [Tags.AGRICULTURE, Tags.CROSS]] # etc`

tnagorra commented on lambda_fns/entry_predict/app.py:
```diff
@@ -0,0 +1,136 @@
+            MessageAttributes=message_attributes
+        )
+    else:
+        print("Message not sent to the processed SQS.")
```
We should use logging module instead of using print

ranjan-stha replied:
logging module used everywhere

tnagorra commented on lambda_fns/entry_predict_input_request/app.py:
```diff
@@ -0,0 +1,53 @@
+                        'StringValue': callback_url
+                    }
+                }
+            } for item in entries_lst]
```
Do we need to keep this inside try block?

ranjan-stha replied:
done

tnagorra commented on lambda_fns/entry_predict_output_request/app.py:
```diff
@@ -0,0 +1,272 @@
+mappings = get_all_mappings()
+categories = get_categories()
+
+fake_data = {
```
do we need this fake_data here?

ranjan-stha replied:
yes, we need the fake sample data for mock server.

tnagorra commented on lambda_fns/entry_predict_output_request/app.py:
```diff
@@ -0,0 +1,272 @@
+            })
+        return all_predictions
+
+    else:
```
We don't need a else if we use return on if block

ranjan-stha replied:
done

tnagorra commented on lambda_fns/entry_predict_output_request/app.py:
```diff
@@ -0,0 +1,272 @@
+
+        return {
+            'statusCode': 200
+        }
```
Should we send 200 when we caught an exception?

ranjan-stha replied:
I guess this should be fine. Because whenever there is an exception, lambda fn retries again and after three failed attempts, the msg is stored in dead letter queue. It won't execute the return statement if there is an error.

tnagorra commented on lambda_fns/extract_docs/app.py:
```diff
@@ -0,0 +1,307 @@
+        bucket_name = bucket_key_split[0]
+        file_path = '/'.join(bucket_key_split[1:])
+        file_name = bucket_key_split[-1]
+        return bucket_name, file_path, file_name
```
We should use path library instead

ranjan-stha replied:
done.

tnagorra commented on lambda_fns/extract_docs/app.py:
```diff
@@ -0,0 +1,307 @@
+
+
+def get_words_count(text):
+    return len(text.split()) if text else 0
```
Should we also use trim here?

ranjan-stha replied:
Trim is actually not required. But I use regex to remove the punctuations to avoid adding them up during count.

tnagorra commented on lambda_fns/extract_docs/app.py:
```diff
@@ -0,0 +1,307 @@
+
+
+def get_extracted_content_links(file_name, mock):
+    with open(os.path.join("/tmp", file_name), "rb") as f:
+        binary = base64.b64encode(f.read())
```
Are we trying to get a temporary file here?

ranjan-stha replied:
@tnagorra : yes, when the request is generated to extract the file contents, the file is first downloaded in the /tmp directory and later it is read from that dir to extract the contents.  Both the downloading and reading process happens one after another, so, there must not be any issue of file being deleted. any possible issue with that?

tnagorra replied:
Python provides a file api to create temporary files. IMO we should use it.

ranjan-stha replied:
done.

ranjan-stha committed:
simplified the mappings list; import fixes; removed a non-reqd. function;

ranjan-stha committed:
logging module used;

ranjan-stha committed:
non-required else removed;

ranjan-stha committed:
try block improvement + code cleanup;

ranjan-stha committed:
updated the Dockerfile;

ranjan-stha committed:
clean on template file;

ranjan-stha committed:
use pathlib library;

ranjan-stha committed:
removed the non-reqd pillars info; stringify the tags version;

ranjan-stha committed:
uses tempfile for temp pdf files; minor code refactoring on text extraction;

ranjan-stha committed:
return in json format;

thenav56 committed:
Add documentation. WIP

Source: https://docs.google.com/document/d/1qnl_z_49VoDN6lLpm0W-VIwuNpqbBEuNcVM5LdZ_3wo/edit

thenav56 committed:
Using poetry

- Use https for the-deep/deepl-pdf-extraction.git

ranjan-stha committed:
added a resource related to bucket acl

ranjan-stha committed:
not storing the lambda pkg in s3;

ranjan-stha committed:
adding the required providers field;

ranjan-stha committed:
setting the version to 3.0

ranjan-stha committed:
removing the s3 bucket acl resource which is not supported by ver 3.0

ranjan-stha committed:
removed a non required comma;

ranjan-stha committed:
updated the entry prediction req handler with lambda and sqs; added for get data from reliability lambda fn

ranjan-stha committed:
lambda layer updated;

ranjan-stha committed:
updated the response format;

ranjan-stha committed:
updated readme.md

ranjan-stha committed:
updated the authoring org to handle list;

ranjan-stha committed:
mappings updated; lambda layer zip file updated;

ranjan-stha committed:
updated req str;

ranjan-stha committed:
updated the request body structure;

ranjan-stha committed:
updated readme.md file;

ranjan-stha committed:
postprocessing of raw predictions;

ranjan-stha committed:
vf_tags endpoint created;

ranjan-stha committed:
attributes added in the mapping;

ranjan-stha committed:
fixed the key name;

ranjan-stha committed:
updated the sample examples;

ranjan-stha committed:
key name updated;

ranjan-stha committed:
round off the prediction value;

ranjan-stha committed:
mapping zip uploaded;

ranjan-stha committed:
endpoint created to retrieve the model info;

ranjan-stha committed:
updated for model_info endpoint;

ranjan-stha committed:
updated mappings

ranjan-stha committed:
updated the zip for mapping;

ranjan-stha committed:
readme updated;

ranjan-stha committed:
updated Readme.md for vf_tags

ranjan-stha committed:
log message updated;

ranjan-stha committed:
removed a sqs body object;

ranjan-stha committed:
req body keys updated;

ranjan-stha committed:
created model_info endpoint for mock

ranjan-stha committed:
model_info data used from lambda fn; queuing batch updates in entry prediction;

ranjan-stha committed:
queuing batch message set to a value;

ranjan-stha committed:
updated mappings;

ranjan-stha committed:
fixes on serialization for model_info;

ranjan-stha committed:
mapping file hashing added;

ranjan-stha committed:
content type added;

ranjan-stha committed:
install deepl parser through requirements.txt;

ranjan-stha committed:
set to 1 page for webpages;

ranjan-stha committed:
content type added for webpage;

ranjan-stha committed:
conditions added to check if clientid is still available;

ranjan-stha committed:
Docker file removed as the parser tool is normally installed now;

ranjan-stha committed:
updated the dev var file;

ranjan-stha committed:
updated vf_tags structure;

ranjan-stha committed:
updated readme.md;

ranjan-stha committed:
mapping key-val updated on vf tags;

ranjan-stha committed:
extract text from a word file(docx, doc) by first converting to a pdf and then extract;

ranjan-stha committed:
ignore the pred values for demographic grps tags;

ranjan-stha committed:
not to generate images and s3 links for images;

ranjan-stha committed:
increase the memory size for text extraction in lambda and set the batchsize to 1 so as to handle each request by a lambda;

ranjan-stha committed:
remove null chars;

ranjan-stha committed:
configs for prod updated;

ranjan-stha committed:
model prediction related sqs, queues updated;

ranjan-stha committed:
timeout handled when extraction is not complete within timeframe; user and system generated msg handled; more support types for text extraction;

ranjan-stha committed:
added configs for creating reserved sqs and lambda for extraction and prediction tags;

ranjan-stha committed:
upgrade option added in the terraform cmd;

ranjan-stha committed:
use alias to provision concurrency  lambda fn that get triggered when requests are received;

ranjan-stha committed:
updating the version of the aws providers;

ranjan-stha committed:
remove profile from config;

ranjan-stha committed:
send only the model info on mockserver;

ranjan-stha committed:
check if key exists in geolocation;

ranjan-stha committed:
function name updated;

ranjan-stha committed:
repo name updated for parser tool

ranjan-stha committed:
provisioned lambda concurrency updated for lambdas;

ranjan-stha committed:
updated the lambda provisioned concurrency;

ranjan-stha committed:
bucket name updated;

ranjan-stha committed:
vpce id updated;
